### PR TITLE
Lb abstaction bug

### PR DESCRIPTION
**Pull request by haim-kermany**
fixing https://github.com/np-guard/vpc-network-config-analyzer/issues/635

### CODE REVIEW

File: pkg/vpcmodel/connectivityAbstraction.go [hunk 1]
```diff
@@ -129,17 +129,24 @@ func (nsa *NodeSetAbstraction) nodeSetAbstractionInformation(mergedConnectivity,
...
+	abstractionInfo.missingEgressConnections = nsa.missingConnections(otherFromNodeSet, mergedConnectivity, nodeSet, fromNodeSet)
+	abstractionInfo.missingIngressConnections = nsa.missingConnections(otherToNodeSet, mergedConnectivity, nodeSet, toNodeSet)
+	abstractionInfo.missingIngressConnections.updateMap(nsa.missingConnections(nodeSetToNodeSet, mergedConnectivity, nodeSet, inNodeSet))
 	return abstractionInfo
 }
 
+type groupDirection int
+
+const (
+	inNodeSet = iota
```
Comment (ShiriMoran) [inline pkg/vpcmodel/connectivityAbstraction.go:141]: rename to `fromToNodeSet`

File: pkg/vpcmodel/connectivityAbstraction.go [hunk 2]
```diff
@@ -72,7 +72,7 @@ func (nsa *NodeSetAbstraction) partitionConnectivityByNodeSet(nodeSet NodeSet) (
 			srcInSet := srcIsNode && slices.Contains(nodeSet.Nodes(), srcNode)
 			dstInSet := dstIsNode && slices.Contains(nodeSet.Nodes(), dstNode)
 			switch {
-			case (!srcInSet && !dstInSet) || conns.isEmpty():
+			case (!srcInSet && !dstInSet):
```
Comment (ShiriMoran) [inline pkg/vpcmodel/connectivityAbstraction.go:75]:
```suggestion
			case !srcInSet && !dstInSet:
```

**Review by ShiriMoran — CHANGES_REQUESTED** [review]

**Comment by adisos** [comment]
@haim-kermany  Please link an issue to the PR 
Please add documentation about the bug on the issue description itself

File: pkg/vpcmodel/connectivityAbstraction.go [hunk 3]
```diff
@@ -129,17 +129,24 @@ func (nsa *NodeSetAbstraction) nodeSetAbstractionInformation(mergedConnectivity,
...
-	abstractionInfo.missingIngressConnections = nsa.missingConnections(otherToNodeSet, mergedConnectivity, nodeSet, true)
-	// nodeSetToNodeSet can be either ingress or egress; as such it suffices to add it to one of them:
-	abstractionInfo.missingIngressConnections.updateMap(nsa.missingConnections(nodeSetToNodeSet, mergedConnectivity, nodeSet, true))
+	abstractionInfo.missingEgressConnections = nsa.missingConnections(otherFromNodeSet, mergedConnectivity, nodeSet, fromNodeSet)
+	abstractionInfo.missingIngressConnections = nsa.missingConnections(otherToNodeSet, mergedConnectivity, nodeSet, toNodeSet)
+	abstractionInfo.missingIngressConnections.updateMap(nsa.missingConnections(nodeSetToNodeSet, mergedConnectivity, nodeSet, inNodeSet))
 	return abstractionInfo
 }
 
+type groupDirection int
```
Comment (adisos) [inline pkg/vpcmodel/connectivityAbstraction.go:138]: add documentation about this type and its values

File: pkg/vpcmodel/connectivityAbstraction.go (hunk 2 above)
Comment (haim-kermany) [reply pkg/vpcmodel/connectivityAbstraction.go]: fixed

File: pkg/vpcmodel/connectivityAbstraction.go (hunk 1 above)
Comment (haim-kermany) [reply pkg/vpcmodel/connectivityAbstraction.go:141]: done

File: pkg/vpcmodel/connectivityAbstration_test.go
Comment (adisos) [inline pkg/vpcmodel/connectivityAbstration_test.go:1]: can this be converted to table-driven tests?

**Review by adisos — CHANGES_REQUESTED** [review]

File: pkg/vpcmodel/connectivityAbstration_test.go
Comment (haim-kermany) [reply pkg/vpcmodel/connectivityAbstration_test.go:1]: done

File: pkg/vpcmodel/connectivityAbstraction.go (hunk 3 above)
Comment (haim-kermany) [reply pkg/vpcmodel/connectivityAbstraction.go:139]: done

File: pkg/vpcmodel/connectivityAbstration_test.go [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+		checkFullConn(nodeSet, outNodes, tt.conn, aConn, info, t)
+	} else {
+		checkMissingConn(nodeSet, outNodes, tt.conn, tt.subConn, aConn, info, t)
+	}
+}
+func TestAbstraction(t *testing.T) {
+	for _, tt := range abstractionTests {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			tt.runTest(t)
```
[the remark this answers is not in the document]
Comment (haim-kermany) [reply pkg/vpcmodel/connectivityAbstration_test.go:59]: fixed

**Review by ShiriMoran — APPROVED** [review]

**Pull request merged**